### PR TITLE
Mongo-project-database

### PR DESCRIPTION
**Pull request by moablomkvist**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -29,7 +39,19 @@ app.use(bodyParser.json())
...
+})
+
+app.get('/:name', (req, res) => {
+  Actor.findOne({name: req.params.name}).then(actor => {
+    if(actor) {
+      res.json(actor)
+    } else {
+      res.status(404).json({ error: 'Not found' })
+    }
+  })
```
Comment (Katsiez) [inline server.js]: Nice first step with setting up the structure!

File: server.js [hunk 2]
```diff
@@ -20,12 +20,46 @@ const Actor = mongoose.model('Actor', {
...
 
-Actor.deleteMany().then(() => {
-  new Actor({ name: 'Nicole Kidman'}).save()
-  new Actor({ name: 'Angelina Jolie'}).save()
+const Movie = mongoose.model('Movie', {
+  title: String,
+  actor: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Actor'
+  }
```
Comment (Katsiez) [inline server.js]: Cool, nicely done!

File: server.js [hunk 3]
```diff
@@ -20,12 +20,46 @@ const Actor = mongoose.model('Actor', {
...
 
 
+//Command: RESET_DATABASE=true npm run dev 
+
+if (process.env.RESET_DATABASE) {
+  console.log('resetting database!')
+
+  const seedDatabase = async () => {
+    await Actor.deleteMany()
+    await Movie.deleteMany()
```
Comment (Katsiez) [inline server.js]: I like that you have included both the Actor and the Movie in seeding the DB.

File: server.js [hunk 4]
```diff
@@ -83,6 +83,25 @@ app.get('/actors', async (req, res) => {
...
+  }
+})
+
+app.get('/actors/:id/movies', async (req, res) => {
+  const actor = await Actor.findById(req.params.id) 
+  if (actor) {
+    const movies = await Movie.find({ actor: mongoose.Types.ObjectId(actor.id) })
+    res.json(movies)
+  } else {
+    res.status(404).json({ error: 'Actor not found' })
```
Comment (Katsiez) [inline server.js]: Nice error handling!

File: server.js [hunk 5]
```diff
@@ -71,6 +71,14 @@ const app = express()
 app.use(cors())
 app.use(bodyParser.json())
 
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else { 
+    res.status(503).json({ error: 'Service not available' })
+  }
+})
```
Comment (Katsiez) [inline server.js:54]: Really good that you thought of different types of errors!

File: server.js [hunk 6]
```diff
@@ -16,48 +18,64 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/movies"
...
+  const populateDatabase = async () => {
+    await Movie.deleteMany();
+  
+    netflixData.forEach(item => {
+      const newMovie = new Movie(item);
+      newMovie.save();
+    })
+  }
+  populateDatabase();
+}
```
Comment (Katsiez) [inline server.js:39]: Neat and clean!

File: server.js [hunk 7]
```diff
@@ -41,54 +28,25 @@ const Movie = mongoose.model('Movie', {
...
   console.log('resetting database!')
   const populateDatabase = async () => {
-    await Movie.deleteMany();
+    await Production.deleteMany();
   
     netflixData.forEach(item => {
-      const newMovie = new Movie(item);
-      newMovie.save();
+      const newProduction = new Production(item);
+      newProduction.save();
```
Comment (Katsiez) [inline server.js:35]: Love this, nice to see that you have challenged yourself to include another route. Props to that.

**Review by Katsiez — COMMENTED** [review]
Moa, well done on your Mongo DB project. You picked a json that had quite a lot of very good detail (movies), which you were able to use to your advantage, especially whilst including 2 different routes. I salute your error handling, that was very well thought-out. 

Overall, nice and clean code. Keep up the good work!

**Review by KarolCislowski — APPROVED** [review]
Nice work! 👍🏻

**Pull request closed without merging**